### PR TITLE
SpiralMatrix solution

### PR DESCRIPTION
**Pull request by pertsodian**
Please help to review. Thanks a lot anh Việt!

### CODE REVIEW

File: week0002/pertsodian/java/src/SpiralMatrixApp.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    SpiralMatrixGenerator matrixGenerator = new SpiralMatrixGenerator();
+    BaseMatrixPrinter matrixPrinter = new ConsoleMatrixPrinter();		
+    SpiralMatrixApp spiralMatrixApp = new SpiralMatrixApp(matrixGenerator, matrixPrinter);
+
+    if (arg.length == 1) {
+      try {
+        int size = Integer.parseInt(arg[0]);
+        if (size < 1)
+          spiralMatrixApp.printUsage();
```
Comment (vietlq) [inline week0002/pertsodian/java/src/SpiralMatrixApp.java]: Defensive programming. Good!

File: week0002/pertsodian/java/src/SpiralMatrixApp.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        else
+          spiralMatrixApp.printSquareSpiralMatrixOfSize(size);
+      }
+      catch (NumberFormatException e) {
+        spiralMatrixApp.printUsage();
+      }
+    }
+    else if (arg.length == 2){
```
Comment (vietlq) [inline week0002/pertsodian/java/src/SpiralMatrixApp.java]: Consistent spacing between `){`

File: week0002/pertsodian/java/src/SpiralMatrixGenerator.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  private int fillLayer(int[][] matrix, int layer, int startElement) {
+    assert matrix.length > 0;
+    assert matrix[0].length > 0;
+
+    int rows = matrix.length;
+    int cols = matrix[0].length;
+
+    for (int i=layer; i<=cols-1-layer; i++)
```
Comment (vietlq) [inline week0002/pertsodian/java/src/SpiralMatrixGenerator.java]: Spacing around operators: `= - + * / <= >= ==` and space after comma `,`

File: week0002/pertsodian/java/src/SpiralMatrixGenerator.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+public class SpiralMatrixGenerator {
+  public int[][] generate(int rows, int cols) {
+    if (rows == 0 || cols == 0)
+      throw new IllegalArgumentException("Matrix rols/cols must be positive");
```
Comment (vietlq) [inline week0002/pertsodian/java/src/SpiralMatrixGenerator.java]: You haven't check for negativity, unless you choose to use unsigned int

File: week0002/pertsodian/java/src/SpiralMatrixGenerator.java (hunk 3 above)
Comment (pertsodian) [inline week0002/pertsodian/java/src/SpiralMatrixGenerator.java]: I personally prefer to read:
`for (int i = layer; i <= cols-1-layer; i++)`

`for (int i = layer; i <= cols - 1 - layer; i++)`
would be too much spacing in my opinion

**Pull request merged**